cmake/public/cuda.cmake

```diff
@@ -216,9 +212,13 @@ endif()
 
 # nvToolsExt
 add_library(torch::nvtoolsext INTERFACE IMPORTED)
-set_property(
-    TARGET torch::nvtoolsext PROPERTY INTERFACE_LINK_LIBRARIES
-    CUDA::nvToolsExt)
+find_path(nvtx3_dir NAMES nvtx3 PATHS "${CUDA_INCLUDE_DIRS}" NO_DEFAULT_PATH)
```

Comment:
Are we not vendoring nvtx this time? I think vendoring would fix #101135, and avoid the need for pytorch/test-infra#4614

Reply:
Almost all users of PyTorch have full installation of CUDA and with nvtx3 installed. I don't want to introduce the burden of maintaining  a third-party library just because test-infra skip it..

Reply:
> Almost all users of PyTorch have full installation of CUDA and with nvtx3 installed.

As per the linked issue it seems nvtx isn't installed by default on windows which should be something to factor in.

> I don't want to introduce the burden of maintaining a third-party library just for passing the CI because test-infra skips it.

Of course if it were only effecting CI then it wouldn't make sense but it seems to be effecting every windows user whose builds PyTorch or a library depending on libtorch.  I also don't think it would introduce any significant maintenance burden. 

> And while nvtx3 is open-sourced, use the particular nvtx3 included with a particular CUDA should be more reasonable.

We already vendor cub so I don't see why it would be unreasonable to vendor nvtx as well.

Reply:
Also note we build nccl which uses it's own version on nvtx as well.

Reply:
> Also note we build nccl which uses it's own version on nvtx as well.

I noticed too. Do you think we use its nvtx a good idea?

Reply:
> Also note we build nccl which uses it's own version on nvtx as well.

If nccl also use its own nvtx3, it seems that it is reasonable to use our own.